llvm/lib/Target/AMDGPU/VOP1Instructions.td

```diff
@@ -898,7 +898,7 @@ multiclass VOP1_Real_NO_DPP_OP_SEL_with_name<GFXGen Gen, bits<9> op,
   VOP1_Real_e32_with_name<Gen, op, opName, asmName>,
   VOP3_Real_with_name<Gen, {0, 1, 1, op{6-0}}, opName, asmName>;
 
-
+let OtherPredicates = [HasFP8ConversionInsts] in {
```

Comment:
Real shall just copy it from pseudo.

Reply:
Okay, I am going to double check  whether we missed something in the pseudo definitions. Thanks

Reply:
This is a little complicate. For _DPP8 and DPP16 reals, only OtherPredicates is copied from pseudo.
However, in VOP3_Real_dpp8_with_name, OtherPredicates is over-written after copied from pseudo:
 OtherPredicates = !if(ps.Pfl.IsRealTrue16, [UseRealTrue16Insts], [TruePredicate])
Can we also keep original OtherPredcates (copied from pseudo) when we add UseRealTrue16Insts?

Reply:
Cc @kosarev @Sisyph

I don't think there's an easy way to _add_ `UseRealTrue16Insts` to `OtherPredicates`.

As a workaround you could set `SubtargetPredicate = HasFP8ConversionInsts` here, instead of using `OtherPredicates`? Or perhaps cleaner we could introduce a new `True16Predicate = ...` for use in `VOP3_Real_dpp8_with_name` to avoid overwriting `OtherPredicates`?

Reply:
Was thinking about adding `True16Predicate` as well. https://github.com/llvm/llvm-project/pull/82245 attempts to implement that.

Reply:
> Cc @kosarev @Sisyph
> 
> I don't think there's an easy way to _add_ `UseRealTrue16Insts` to `OtherPredicates`.
> 
> As a workaround you could set `SubtargetPredicate = HasFP8ConversionInsts` here, instead of using `OtherPredicates`? Or perhaps cleaner we could introduce a new `True16Predicate = ...` for use in `VOP3_Real_dpp8_with_name` to avoid overwriting `OtherPredicates`?

Do you think listconcat works for OtherPredicates to add an additional predicate?

SubtargetPredicate does not work for two reasons"
1) SubtargetPredicate is not copied from pseudo to real for dpp8 and dpp16, which means SubtargetPrediacte will get lost (unused) in real.
2) We have the same issue to over-write SubtargetPredicate (Base_VOP3_DPP16)
   let SubtargetPredicate = HasDPP16;

Reply:
> Was thinking about adding `True16Predicate` as well. #82245 attempts to implement that.

Can you explain why listconcat does not work for OtherPredicates to add an additional predicate?
I think it should work, and we don;t have to introduce a new predicate.

Reply:
Discussed in https://github.com/llvm/llvm-project/pull/82245#discussion_r1494929918.